presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -2125,4 +2127,53 @@ core::PlanFragment VeloxQueryPlanConverter::toVeloxQueryPlan(
         toJsonString(partitioningHandle));
   }
 }
+
+velox::core::PlanFragment VeloxQueryPlanConverter::toBatchVeloxQueryPlan(
+    const protocol::PlanFragment& fragment,
+    const std::shared_ptr<protocol::TableWriteInfo>& tableWriteInfo,
+    const protocol::TaskId& taskId,
+    const std::shared_ptr<operators::ShuffleInterface>& writerShuffle) {
+  auto planFragment = toVeloxQueryPlan(fragment, tableWriteInfo, taskId);
+
+  // If the last node is a PartitionedOutputNode, it means this fragment ends
+  // with a shuffle stage. We convert the PartitionedOutputNode to a chain of
+  // following nodes:
+  // (1) A PartitionAndSerializeNode.
+  // (2) A "gather" LocalPartitionNode that gathers results from multiple
+  //     threads to one thread.
+  // (3) A ShuffleWriteNode.
+  if (planFragment.planNode->name() != "PartitionedOutput") {
+    return planFragment;
+  }
+  auto partitionedOutputNode =
+      dynamic_pointer_cast<const core::PartitionedOutputNode>(
+          planFragment.planNode);
+  VELOX_CHECK_NOT_NULL(partitionedOutputNode);
+
+  auto partitionAndSerializeNode =
+      std::make_shared<operators::PartitionAndSerializeNode>(
+          "shuffle-partition-serialize",
```

Comment:
Plan node IDs in Presto are numbers. Consider using the same.

Reply:
I see PartitionedOutputNode hardcoding "root" as plan node id. What is the convention here in Velox context?